src/hooks/useOwner.ts

```diff
@@ -0,0 +1,37 @@
+//- by Alejo Viola and Leo
+
+import React, { useEffect, useState } from "react";
+import { TokenVestingController } from "contracts/types";
+
+//- Web3 Imports
+import { useWeb3React } from '@web3-react/core';
+import { Web3Provider } from '@ethersproject/providers/lib/web3-provider';
+
+enum OwnerState {
+  Disconnected = 0,
+  Autorized = 1,
+  NoAuthorized = 2,
+}
+
+export const useOwner = () => {
+
+  //- Wallet Data
+  const walletContext = useWeb3React<Web3Provider>();
+  const { account } = walletContext;
+
+  const OWNER = '0x728f7ee36Fb640535327b2b1C2695854998053A8';
```

Comment:
No hardcoded variables, also ensure that the variable name follows proper naming conventions.

Reply:
We can't implement the owner check properly until we solve something in the interface